### PR TITLE
Feature/tt 610 sdk bnb

### PR DESCRIPTION
**Pull request by jhammer-tatum**

### CODE REVIEW

File: packages/blockchain/bnb/src/lib/services/bnb.tx.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        ],
+      },
+    }
+  }
+
+  const sendTransaction = async (
+    body: TransferBnb,
+    options: { testnet: boolean; token?: string; issuerAccount?: string },
+  ) => {
+    return ApiServices.blockchain.bnb.bnbBroadcast({
```
Comment (samuelsramko) [inline packages/blockchain/bnb/src/lib/services/bnb.tx.ts:65]: you should wrap with if (body.signatureId) ApiServices.blockchain.bnb.sendTransaction) else this

File: packages/blockchain/bnb/src/lib/services/bnb.tx.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    } catch (e) {
+      if ((e as any).code === SdkErrorCode.INSUFFICIENT_FUNDS) {
+        throw e
+      }
+      throw new BnbSdkError(SdkErrorCode.TX_PREPARATION_FAILED)
+    }
+  }
+
+  return {
+    prepareTransaction,
```
Comment (samuelsramko) [inline packages/blockchain/bnb/src/lib/services/bnb.tx.ts:131]: missing JS Doc with texts and descriptions

**Review by samuelsramko — CHANGES_REQUESTED** [review]

File: packages/blockchain/bnb/src/lib/services/bnb.tx.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+  /**
+   * Send BNB transaction to the blockchain. This method broadcasts signed transaction to the blockchain.
+   * @param body content of the transaction to broadcast
+   * @param options mainnet or testnet version
+   * @returns transaction id of the transaction in the blockchain
+   */
+  const sendTransaction = async (body: TransferBnb, options: { testnet: boolean }) => {
+    if (body.signatureId) {
+      return ApiServices.blockchain.bnb.bnbBlockchainTransfer(body as any)
```
Comment (alexloiko) [inline packages/blockchain/bnb/src/lib/services/bnb.tx.ts:82]: please do not use cast to any

File: examples/bnb-example/src/app/bnb.tx.example.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { TatumBnbSDK } from '@tatumio/bnb'
+import { Currency } from '@tatumio/api-client'
+
+export async function bnbTxExample() {
+  const bnbSDK = TatumBnbSDK({ apiKey: 'PLACE-YOUR-API-KEY-HERE' })
```
Comment (alexloiko) [inline examples/bnb-example/src/app/bnb.tx.example.ts:5]: we've discussed to place this key everywhere '75ea3138-d0a1-47df-932e-acb3ee807dab' in examples

File: examples/bnb-example/src/app/bnb.wallet.example.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { TatumBnbSDK } from '@tatumio/bnb'
+
+export async function bnbWalletExample() {
+  const bnbSDK = TatumBnbSDK({ apiKey: '75ea3138-d0a1-47df-932e-acb3ee807dab' })
+  const wallet = bnbSDK.wallet
+
+  const { address, privateKey } = await wallet().generateWallet(true)
+  console.log(` My public address is ${address}, with private key ${privateKey}.`)
```
Comment (alexloiko) [inline examples/bnb-example/src/app/bnb.wallet.example.ts:8]: minor. excessive space in comment

File: packages/api-client/src/generated/core/OpenAPI.ts [hunk 6]
```diff
@@ -1,10 +1,10 @@
 /* istanbul ignore file */
 /* tslint:disable */
 /* eslint-disable */
-import type { ApiRequestOptions } from './ApiRequestOptions';
+import type { ApiRequestOptions } from './ApiRequestOptions'
```
Comment (alexloiko) [inline packages/api-client/src/generated/core/OpenAPI.ts:4]: changes in generated files

File: packages/blockchain/bnb/src/lib/services/bnb.kms.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  fromPrivateKey: string,
+  testnet: boolean,
+  provider?: string,
+) => {
+  if (tx.chain !== Currency.BNB) {
+    throw new BnbSdkError(SdkErrorCode.KMS_CHAIN_MISMATCH)
+  }
+  const bnbClient = new BncClient(
+    provider
+      ? provider
```
Comment (alexloiko) [inline packages/blockchain/bnb/src/lib/services/bnb.kms.ts:26]: please avoid double chaining of in-place ifs

File: packages/blockchain/bnb/src/lib/services/bnb.tx.ts [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        throw new BnbSdkError(
+          SdkErrorCode.INSUFFICIENT_FUNDS,
+          `Insufficient funds to create transaction from sender account ${account} -> available balance is ${accountBalance.toString()}, required balance is ${requiredBalance.toString()}.`,
+        )
+      }
+      bnbClient.setAccountNumber(account.account_number || 0)
+      const signedTx = await bnbClient._prepareTransaction(msg, signMsg, fromAddress, account.sequence, memo)
+      return signedTx.serialize()
+    } catch (e) {
+      if ((e as any).code === SdkErrorCode.INSUFFICIENT_FUNDS) {
```
Comment (alexloiko) [inline packages/blockchain/bnb/src/lib/services/bnb.tx.ts:168]: please avoid cast to any

**Pull request merged**